### DISCUSSION TITLE
Stored procedures as a postgresql n00b

### DISCUSSION THREAD

**Discussion by noconsulate** [supabase/supabase#1107] [Questions] [answered]
I've got my client inserting, reading, updating, and most importantly: subscribing. Everything works really well and I'm surprised how easy it is. Thanks to a poster here I'm adapting my data model to use tables properly rather than relying on a JSON column. What was essentially a two-dimensional array of JSON will now be broken up into rows in a table. 

I could naively use javascript on the client to create the rows, either one cell at a time or perhaps I can do it row by row. Either way I'll have to make several calls to the database where if I make an RPC I'm sure I could do it with one call. The problem is I don't know how to do RPCs! 

I've spent time researching this and stored procedures indeed seems to be an advanced topic. I've brushed up on the basics of postgresql but it seems like I'd have to do a lot of studying to be able to do this. PLV8 offers to make writing the logic of a procedure a lot easier but without fundamentals I'm still at a loss. 

My question is should I bother with an RPC? I doubt it would mean any difference in performance, it just seems handy and would spare a few lines of code in my client. Or perhaps there is a guide somewhere that would bring me up to speed and enable me to achieve this. God knows it's hard to find anything good on Google these days. 

tl-dr: How can I implement an RPC without being a postgresql expert?

**Comment by silentworks** [comment, marked as answer]
Yes a stored procedure might be better but if you have it working without needing one you can work that way for now. I'm no Postgres expert myself, I do however have a few years of experience with MySQL so that helps a bit. I learn a lot about Postgres from this website https://www.postgresqltutorial.com/, it has been helpful in understanding some of the concepts that I was not familiar with.

**Reply by noconsulate** [reply]
Yes I'm sure your experience with MySQL counts for a lot! I did see that tutorial and it seems... long. And the RCP stuff isn't until the very end! Maybe I'll start into it and see how it goes.

**Reply by silentworks** [reply]
If you have any questions just ask here and I'm sure someone will help. Supabase is relatively a young project so I'm sure the documentation will get better over time.

**Comment by noconsulate** [comment]
I just want to follow up for posterity. The plpgsql [tutorial](https://www.postgresqltutorial.com/postgresql-plpgsql/) at the site @silentworks recommended got me up and running in no time. Well, about a day. I didn't need any advanced knowledge of postgresql to understand it, just the basic queries. It wasn't hard to create an RCP that does everything I need. You just need to know how to do the required SQL operations, how the functions are stored and accessed, and then learn the straight forward syntax.
